### DISCUSSION TITLE
Introduce a new Filters block

### DISCUSSION THREAD

**Discussion by shaunandrews** [woocommerce/woocommerce#44955] [WooCommerce, Blocks, and Site Editing]
Right now, our product filter blocks are disconnected from our upcoming Product Collection block. This mean you can add a filter-type block anywhere in a template.

<img width="350" alt="CleanShot 2023-08-22 at 14 52 07@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/86696540-6ccc-4f6a-9cb3-6882de388380">

This is a little strange, and relies heavily on using the default query as a way to connect the filters to the products shown. It also makes it hard to connect these filters with a specific Product Collection block if there are multiple collections shown.

I think it might make sense to follow the pattern of the WordPress Core Buttons and Button blocks; We'd create a new **Collection Filters** block, which would contain inner blocks for each filter.

<img width="1796" alt="CleanShot 2023-08-22 at 15 33 35@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/59ab08da-a08d-4999-94c1-742a2521c3e1">

**Comment by nerrad** [comment]
Presumably, this could also allow for different layout/orientation options for the filters as well (controlled via the collection filters layout block)?

**Reply by shaunandrews** [reply]
Something like this?

<img width="427" alt="CleanShot 2023-08-22 at 16 14 34@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/95a48944-4b4c-4ea3-bfef-04af2bce6dca">

<img width="1512" alt="CleanShot 2023-08-22 at 16 18 00@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/c0c098c1-283a-45ef-9da4-2146d174264c">

I think that could come with patterns available to the root Collection block?

**Reply by roykho** [reply]
Even with the proposed new Filter Collection block, we can still achieve different layouts. We can still add columns within the Product Collection root block and just add the Filter Collection block to the designated column. And on top of this we could have pre-defined layouts in the Product Collection root as well as mentioned.

**Reply by nerrad** [reply]
> Something like this?

Yup. I realize there are multiple ways this can be implemented. I was mostly wondering whether the container block should allow for the orientation of the child block layout (via layout controls). Or is the plan to simply have the child filter blocks be arranged responsively within the container automatically (dependent on its width)?

**Reply by shaunandrews** [reply]
My expectation is that these blocks can be arranged any ways possible using groups (row, stack) and columns, and not necessarily tied to any particular settings on the Filters block.

**Reply by imanish003** [reply]
Hi @shaunandrews @roykho,

I'm curious about the necessity of having the `Collection Filters` block serve as the parent/ancestor of the Filter blocks. Is this choice influenced by specific technical constraints or considerations?

From my understanding, I had envisioned the ability to integrate Filter blocks directly within the `Product Collection` block, with the necessary context being provided by the Product Collection itself. This way, we could use layout blocks such as `Row` and `Column` to arrange the filter blocks in a desired manner, anywhere inside `Product Collection` block.

Could you provide some insight if I might be overlooking any important factors in this approach? Your input would be greatly appreciated!

CC: @kmanijak

**Reply by roykho** [reply]
This is just from me playing around with the wrapper container locally. It makes it easier to have predefined style. For example if we have predetermined that the filters are usually either horizontal or vertical, we can already have HTML/CSS in place that accommodates for that. Without the container, then the user has to create this layout themselves by adding columns/rows...etc.

> This way, we could use layout blocks such as Row and Column to arrange the filter blocks in a desired manner, anywhere inside Product Collection block.

They can still choose to put it into specific column/rows themselves even with the wrapper container.

**Reply by shaunandrews** [reply]
My intention is that the Collection Filters block could still allow for other layout-type blocks like Group.

My thinking behind this new block is that Filters are often grouped together both within the markup and in the visual rendering. With this new wrapper container, we can make it easy to visual style our grouped filters — and benefit by having interactions (adding, moving, styles, and settings) relegated to a single block. It might make sense to surface some individual filter settings to our new container block as well. This could make it easier to manage filters without having to select each individual block.

**Reply by dinhtungdu** [reply]
And if the merchants ever want a filter submit button, they can have only one button for all filters.

**Reply by imanish003** [reply]
Understood, and thank you everyone for explaining the rationale behind it.  🙌🏻

**Reply by shaunandrews** [reply]
> And if the merchants ever want a filter submit button, they can have only one button for all filters.

Could you elaborate here? I don't quite understand your point. Would this be a regression or an improvement? What about this decisions precludes us from allowing multiple buttons?

**Reply by dinhtungdu** [reply]
@shaunandrews This comes from the current filter blocks. Each block has an option to show a Filter Submit button. So the site can end up have multiple filter buttons. To me, it's better to have a single Filter Submit button to submit the current choices from all filters.

I say ` if the merchants ever want a filter submit button` because with Interactivity API, the transition is quite seamless, which lowers significantly the need of a filter submit button.

**Reply by shaunandrews** [reply]
Ah I see. We could still offer multiple buttons, but they would probably just submit all the filters at once. As you said, multiple buttons probably isn’t the best UX anyways.

**Comment by roykho** [comment]
Thanks for putting up this discussion. It will definitely help with our efforts here https://github.com/woocommerce/woocommerce-blocks/discussions/10678

With that said, I agree with this simplified design and to consolidate the filters into a new filters block which can only be added in the context of the product collection block.

Once this is decided, I am happy to add this task into our Epic https://github.com/woocommerce/woocommerce-blocks/issues/10566

**Comment by kmanijak** [comment]
Can I confirm that assumes 1-to-1 relation between Collection Filters and Product Collection? So basically there can be multiple Product Collection blocks on a page and each of them may (or may not) have their own Collection Filters. And I assume it doesn't matter if Product Collection inherits the query from a template or not - it can have Collection Filters.

That's how I interpret these designs but wanted to verbalize my assumptions to make sure we capture all of the requirements. 🙌 

BTW I like that idea. Recently we had a discussion about that topic (with that exact idea included) with @imanish003 and @dinhtungdu, so thank you for covering that!

**Reply by dinhtungdu** [reply]
> And I assume it doesn't matter if Product Collection inherits the query from a template or not - it can have Collection Filters.

This is my thought too. Filters shouldn't (and don't need to) know whether Product Collection inherits query from template or not. They should work with any product collections that have valid query.

**Reply by shaunandrews** [reply]
Yup, that maps to my thinking.
